swap/src/bin/asb.rs

```diff
@@ -45,23 +45,49 @@ const DEFAULT_WALLET_NAME: &str = "asb-wallet";
 
 #[tokio::main]
 async fn main() -> Result<()> {
-    let opt = Arguments::from_args();
-    asb::tracing::init(LevelFilter::DEBUG, opt.json).expect("initialize tracing");
-
-    let config_path = if let Some(config_path) = opt.config {
+    let Arguments {
+        testnet,
+        json,
+        config,
+        cmd,
+    } = Arguments::from_args();
+    asb::tracing::init(LevelFilter::DEBUG, json).expect("initialize tracing");
+
+    let config_path = if let Some(config_path) = config {
         config_path
+    } else if testnet {
+        env::Testnet::getConfigFileDefaults()?.config_path
     } else {
-        default_config_path()?
+        env::Mainnet::getConfigFileDefaults()?.config_path
     };
 
     let config = match read_config(config_path.clone())? {
         Ok(config) => config,
         Err(ConfigNotInitialized {}) => {
-            initial_setup(config_path.clone(), query_user_for_initial_testnet_config)?;
+            initial_setup(config_path.clone(), query_user_for_initial_config, testnet)?;
             read_config(config_path)?.expect("after initial setup config can be read")
         }
     };
 
+    let env_config = if testnet {
+        env::Testnet::get_config()
+    } else {
+        env::Mainnet::get_config()
+    };
+
+    if config.monero.network != env_config.monero_network {
+        bail!(format!(
+            "Expected monero network in config file to be {:?} but was {:?}",
+            env_config.monero_network, config.monero.network
+        ));
+    }
+    if config.bitcoin.network != env_config.bitcoin_network {
+        bail!(format!(
+            "Expected bitcoin network in config file to be {:?} but was {:?}",
+            env_config.bitcoin_network, config.bitcoin.network
+        ));
+    }
```

Comment:
We are overloading `main.rs` quite a bit here. All of this "derivation" of data based of the arguments could easily live in a different function that can be nicely tested even based on actual commandline parameters like we did here (this never merged): https://github.com/comit-network/xmr-btc-swap/blob/948cc99e9d87dfd710fa3cae15bb777710669dbd/swap/src/cli/command.rs#L213-L269

Reply:
Will tackle with in follow for both CLI and ASB.

Reply:
After that bit me: https://github.com/comit-network/xmr-btc-swap/pull/490/commits/9f5a686d4897797f1fd496dd30499ea165334657